scipy/spatial/transform/tests/test_rotation.py

```diff
@@ -39,6 +39,52 @@ def test_from_single_2d_quaternion():
     assert_array_almost_equal(r.as_quat(), expected_quat)
 
 
+def test_from_quat_scalar_first():
+    rng = np.random.RandomState(0)
+
+    r = Rotation.from_quat([1, 0, 0, 0], scalar_first=True)
+    assert_allclose(r.as_matrix(), np.eye(3), rtol=1e-15, atol=1e-16)
+
+    r = Rotation.from_quat(np.tile([1, 0, 0, 0], (10, 1)), scalar_first=True)
+    assert_allclose(r.as_matrix(), np.tile(np.eye(3), (10, 1, 1)),
+                    rtol=1e-15, atol=1e-16)
+
+    q = rng.randn(100, 4)
```

Comment:
I'm not really following why all these extra tests are needed? Would think that you could just compare e.g.
`
assert(Rotation.from_quat([1, 2, 3, 4], scalar_first=True).as_quat(scalar_first=False), Rotation.from_quat([2, 3, 4, 1], scalar_first=False).as_quat(scalar_first=False))`

Reply:
Perhaps I've lost some skills in writing tests.

My reasoning was to keep tests as "unit" as possible and not couple new features in `as_quat` and `from_quat`.

Maybe it's an overkill here, but intuitively it makes sense to me. For example, what if I not implement anything for `scalar_first`? Then `as_quat` and `from_quat` will work consistently, but incorrectly.

Reply:
I guess I've mostly missed your point.

Let me answer properly. The first pair of my tests are checking how identity quaternion looks in scalar-first format. 

The second pair works on a random input. And in this case using `np.roll` is inevitable.

I think my tests are fine, but I might have just used 2 simpler tests as your suggested (for a single and multiple quaternions).

Reply:
Got it, yeah I think the tests are fine as is